torch/_inductor/codegen/cpp_prefix.h

```diff
@@ -66,21 +68,39 @@ struct IsVecType: std::false_type {};
 #if INDUCTOR_USE_VECTOR_TYPES()
 template <typename T>
 struct IsVecType<at::vec::Vectorized<T>>: std::true_type {};
+template <typename T, int N>
+struct IsVecType<at::vec::VectorizedN<T, N>>: std::true_type {};
 #endif
 
 template <typename T>
-struct WeightRecp {
-  using scalar_t = typename T::value_type;
-  std::vector<scalar_t> weight_recps;
-  WeightRecp(uint64_t N) {
-    weight_recps.reserve(N);
-    for (const auto i : c10::irange(N)) {
-      weight_recps.push_back(
-          scalar_t(static_cast<double>(1) / static_cast<double>(i + 1)));
-    }
+struct WelfordHelper {
+  // A data struct to help welford reduction:
+  // 1. Save the reciprocal of weights to avoid redundant divisions.
+  // 2. Save the welford stack, which is used to combine welford reduction
+  //    with cascade summation to improve numerical stability.
+  static std::vector<typename T::value_type> weight_recps;
+  std::vector<Welford<T>> welford_stk;
+  uint64_t depth; // depth of welford_stk.
+  uint64_t num_chunks; // number of chunks stored in welford_stk.
+  WelfordHelper() {}
+  WelfordHelper(uint64_t N) {
+    uint64_t m = (N + kChunkSize - 1) / kChunkSize; //div up
```

Comment:
The concern is that we define this magic number twice both in the `cpp.py` and `cpp_prefix.h`. If we want to change this magic number in the future, how could we ensure both of these 2 places are changed in time? Could `kChunkSize` be part of `WelfordHelper`, and it can passed from python when create `WelfordHelper`?

Reply:
Done, thanks for your comment!